server/db/index.js

```diff
@@ -1,8 +1,24 @@
 const mongoose = require("mongoose")
+const { createClient } = require("redis")
 class InitiateDB {
     connect(uri) {
         return mongoose.connect(uri)
     }
+
+    connectRedis(url){
+        const clint = createClient({
+            url
+        })
+        clint.on("error", (err) => {
+            throw new Error(err)
+        })
+
+        clint.connect().then((param) => {
+            console.log("redis database connected successfully")
+        }).catch((err) => {
+            console.log("failed to connect redis")
+        })
+    }
```

Comment:
You can add a condition in this code. It will help you to write error-free code

you can write like this:

    `  connectRedis(url){
        if(url){
            const clint = createClient({
                url
            })
            clint.on("error", (err) => {
                throw new Error(err)
            })
    
        clint.connect().then((param) => {
            console.log("redis database connected successfully")
        }).catch((err) => {
            console.log("failed to connect redis")
        })
    }else {
        console.log("invalid url. please provide valid url")
    }
   }
`